hedera-node/hedera-app/src/test/java/com/hedera/node/app/blocks/BlockRetentionManagerTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2024-2025 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.blocks;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.hedera.node.app.blocks.impl.FileBlockItemWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class BlockRetentionManagerTest {
+    @TempDir
+    private Path uploadedDir;
+
+    private BlockRetentionManager blockRetentionManager;
+
+    private static final int AWAIT_SECONDS_TIMEOUT = 60;
+
+    @Test
+    void testCleanupShouldNotRunUntilScheduleTime() throws IOException {
+        final long retentionPeriodMs = 10;
+        final long cleanupPeriodMs = 100;
+        blockRetentionManager = new BlockRetentionManager(
+                uploadedDir, Duration.ofMillis(retentionPeriodMs), Duration.ofMillis(cleanupPeriodMs), 4);
+        final Runnable cleanupTask = () -> blockRetentionManager.startCleanup();
+        blockRetentionManager.scheduleRepeating(cleanupTask);
+
+        final var blockFileName = "file1" + "." + FileBlockItemWriter.RECORD_EXTENSION;
+
+        createTestFile(blockFileName);
```

Comment:
These unit tests are creating files with the same filename, meaning we won't be able to run these tests in parallel. If we use `Files.createTempFile(...)`, then it will loop until it finds a (unique) random filename:

```
// Using `RECORD_EXTENSION` as the `suffix` argument should ensure the file has the 
// right extension
Path blockFileName = Files.createTempFile(uploadedDir, "test", FileBlockItemWriter.RECORD_EXTENSION);
````

Reply:
Makes sense, updated.